ext/RMagick/rmimage.c

```diff
@@ -2611,6 +2611,11 @@ set_profile(VALUE self, const char *name, VALUE profile)
     exception = AcquireExceptionInfo();
     m = GetMagickInfo(name, exception);
     CHECK_EXCEPTION()
+    if (m == NULL)
+    {
+        (void) DestroyExceptionInfo(exception);
+        rb_raise(Class_ImageMagickError, "unknown image profile name");
```

Comment:
This should raise `ArgumentError` as an exception...., like

https://github.com/rmagick/rmagick/blob/f738055446798ca5b370a2b5bedc667d8f7a5089/ext/RMagick/rmimage.c#L6511-L6519

Reply:
Changed